articles/proofs-arguments-and-zero-knowledge/chapter_19.md

```diff
@@ -0,0 +1,174 @@
+---
+Title: Bird’s Eye View of Practical Arguments
+Status: In Progress
+Level: "6"
+---
+# Chapter 19: Bird’s Eye View of Practical Arguments
+
+![](attachments/taxonomy.png)
+
+## 19.1 A Taxonomy of SNARKs
+
+Outside of the **linear-PCP-based SNARKs**, most known **SNARKs** are obtained by combining some **IP**,  **MIP**, or **constant-round polynomial IOP** with a **polynomial commitment** scheme.
+
+For example:
+
+| Polynomial IOP approaches      | Polynomial Commitment Approaches                                                     | SNARK                                            |
+| ------------------------------ | ------------------------------------------------------------------------------------ | ------------------------------------------------ |
+| IPs                            | FRI-based (multilinear) polynomial commitments                                       | Virgo                                            |
+| IPs                            | Discrete-log-based (multilinear) polynomial commitments (Bulletproofs, Hyrax-commit) | Hyrax                                            |
+| IPs                            | KZG-based (multilinear) polynomial commitments                                       | zk-vSQL and Libra                                |
+| MIPs                           | Multilinear polynomial commitments                                                   | Spartan, Xiphos, Kopis, Brakedown, and Shockwave |
+| Constant-round polynomial IOPs | FRI-based (univariate) polynomial commitments                                        | Aurora, Fractal, and Redshift                    |
+| Constant-round polynomial IOPs | KZG-based (univariate) polynomial commitments                                        | Marlin                                           |
+| Constant-round polynomial IOP  | Ligero                                                                               | Ligero                                           |
+
+Beside that, the most popular variant of the SNARK derived from GGPR’s linear PCP is
```

Comment:
linear PCP combine with pairing

Reply:
Actually, GGPR's linear PCP is more overview

Reply:
more overview ?

Reply:
I mean more overall 👍